### PR TITLE
Integrate leader-election feature with etcd-druid.

### PR DESCRIPTION
**Pull request by ishan16696**
**What this PR does / why we need it**:
It integrates the [leader-election feature](https://github.com/gardener/etcd-backup-restore/pull/353) with etcd-druid.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
To merge only after etcd-backup-restore `v0.15.0` release

**Release note**:
```other operator
Leader-election specifications can be configured through Etcd resource spec via `.spec.backup.leaderElection`.
```

### CODE REVIEW

**Comment by ishan16696** [comment]
@timuthy can you review this PR also ... we can merge this after backup-restore release `v0.15.0`

File: charts/etcd/templates/etcd-statefulset.yaml
Comment (aaronfern) [inline charts/etcd/templates/etcd-statefulset.yaml:223]:
```suggestion
        {{- if .Values.backup.leaderElection.reelectionPeriod }}
        - --etcd-connection-timeout-leader-election={{ .Values.backup.leaderElection.etcdConnectionTimeout }}
        {{- end }}
        {{- if .Values.backup.leaderElection.etcdConnectionTimeout }}        
        - --reelection-period={{ .Values.backup.leaderElection.reelectionPeriod }}
        {{- end }}
```
These two values are interchanged

File: charts/images.yaml
Comment (aaronfern) [inline charts/images.yaml:5]: Please remember to revert this before merging

**Review by aaronfern — CHANGES_REQUESTED** [review]

File: charts/etcd/templates/etcd-statefulset.yaml
Comment (ishan16696) [reply charts/etcd/templates/etcd-statefulset.yaml]: thanks for catching this.

**Review by aaronfern — APPROVED** [review]
Thanks for the changes!
/lgtm

File: charts/images.yaml
Comment (ishan16696) [reply charts/images.yaml]: done

**Pull request merged**
